### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by Andarwinux**

### CODE REVIEW

**Comment by shinchiro** [comment]
https://github.com/shinchiro/mpv-winbuild-cmake/commit/4ba1e56e642e5c368fba50465358426e4464e549

**Comment by shinchiro** [comment]
Modify pkgconfig in BUILD_COMMAND step

**Comment by Andarwinux** [comment]
> [4ba1e56](https://github.com/shinchiro/mpv-winbuild-cmake/commit/4ba1e56e642e5c368fba50465358426e4464e549)

https://github.com/FFmpeg/FFmpeg/commit/2f24f10d9cf34ddce274496c4daa73f732d370c1

> Modify pkgconfig in BUILD_COMMAND step

This will destroy the read-only nature of the source, not a good idea

**Comment by shinchiro** [comment]
Modify the pkgconfig in <BINARY_DIR> before it get installed. Not the <SOURCE_DIR>

**Comment by Andarwinux** [comment]
> Modify the pkgconfig in <BINARY_DIR> before it get installed. Not the <SOURCE_DIR>

done.

File: packages/openal-soft.cmake [hunk 1]
```diff
@@ -19,6 +19,7 @@ ExternalProject_Add(openal-soft
         -DALSOFT_BACKEND_PIPEWIRE=OFF
     BUILD_COMMAND ${EXEC} ninja -C <BINARY_DIR>
```
Comment (shinchiro) [inline packages/openal-soft.cmake:20]: Modify .pc at this step

File: packages/libmodplug.cmake [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    INSTALL_COMMAND ${MAKE} install
-    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND ${EXEC} CONF=1 cmake -H<SOURCE_DIR> -B<BINARY_DIR>
+        -G Ninja
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE}
+        -DCMAKE_INSTALL_PREFIX=${MINGW_INSTALL_PREFIX}
+        -DCMAKE_FIND_ROOT_PATH=${MINGW_INSTALL_PREFIX}
+        -DBUILD_SHARED_LIBS=OFF
+    BUILD_COMMAND ${EXEC} ninja -C <BINARY_DIR>
```
Comment (shinchiro) [inline packages/libmodplug.cmake:14]: Do echo Cflags.private here. Rather than putting new file

Comment (Andarwinux) [reply packages/libmodplug.cmake:14]: If is win32, libmodplug cmake won't install pkgconfig, and while it's possible to fix this with a patch, it defeats the purpose of my switching it to cmake, which is to keep the source tree read-only.

**Comment by shinchiro** [comment]
> [FFmpeg/FFmpeg@2f24f10](https://github.com/FFmpeg/FFmpeg/commit/2f24f10d9cf34ddce274496c4daa73f732d370c1)

The fix is incomplete. `src/libavfilter/vsrc_testsrc_vulkan.c` file need to fix too

```
src/libavfilter/vsrc_testsrc_vulkan.c:234:47: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'VkSampler' (aka 'unsigned long long') [-Wint-conversion]
  234 |                                               NULL, &s->opts, sizeof(s->opts));

```

**Pull request merged**